### PR TITLE
refactor(github): Simplify datasource cacheability detection

### PR DESCRIPTION
**Pull request by zharinov**
## Context

- Ref: #18188

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/util/github/graphql/datasource-helper.ts [hunk 1]
```diff
@@ -256,7 +260,7 @@ export class GithubGraphqlDatasourceHelper<
     return resultPromise;
   }
 
-  private async getItems(): Promise<ResultItem[]> {
+  public async getItems(): Promise<ResultItem[]> {
```
Comment (viceice) [inline lib/util/github/graphql/datasource-helper.ts:263]:
```suggestion
  async getItems(): Promise<ResultItem[]> {
```

File: lib/util/github/graphql/datasource-helper.spec.ts [hunk 2]
```diff
@@ -382,5 +385,36 @@ describe('util/github/graphql/datasource-helper', () => {
...
+            .scope('https://api.github.com/')
+            .post('/graphql')
+            .reply(200, resp(data, undefined, isPrivate));
+
+          const instance = new GithubGraphqlDatasourceHelper(
+            { packageName: 'foo/bar' },
+            http,
+            adapter
+          );
+          await instance.getItems();
```
Comment (viceice) [inline lib/util/github/graphql/datasource-helper.spec.ts:413]: can we also expect somthing on the retun value, just in case?

Comment (zharinov) [reply lib/util/github/graphql/datasource-helper.spec.ts:413]: I think it's out of scope for this particular tests, but we do this in previous ones.

**Review by rarkins — COMMENTED** [review]
Is this fixing or just refactoring?

**Review by rarkins — APPROVED** [review]

**Review by viceice — APPROVED** [review]

**Pull request merged**